legend-engine-authentication-pure/src/main/resources/core_authentication/authentication/metamodel.pure

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import meta::pure::authentication::*;
+import meta::pure::credential::*;
+
+Class <<typemodifiers.abstract>> meta::pure::credential::Credential
+{
+  provider: CredentialProvider[1];
+}
+
+Class meta::pure::credential::PlainTextCredential extends Credential
+{
+}
+
+Class meta::pure::credential::PrivateKeyCredential extends Credential
+{
+}
+
+Class meta::pure::credential::JsonWebTokenCredential extends Credential
+{
+}
+
+Class <<typemodifiers.abstract>> meta::pure::credential::CredentialProvider
+{
+}
+
+Class meta::pure::credential::VaultCredentialProvider extends CredentialProvider
+{
+  vaultReference: String[1];
+}
+
+Class meta::pure::credential::OidcCredentialProvider extends CredentialProvider
+{
+  discoveryUrl: String[1];
+}
+
+Class meta::pure::credential::OAuthTokenProvider extends CredentialProvider
+{
+  url: String[1];
+}
+
+Class <<typemodifiers.abstract>> meta::pure::authentication::Authentication
+{
+}
+
+Class meta::pure::authentication::UsernamePasswordAuthentication extends Authentication
```

Comment:
```Authentication``` types have 2 classes of attributes.  
* Attributes such as username etc that are not sensitive and can be in-lined in the model 
* Attributes such as credentials which are sensitive. These cannot be in-lined in the model and therefore we have a ```Credential``` type that describes where and how these values can be obtained